concrete-core/src/math/random/generator.rs

```diff
@@ -213,56 +214,108 @@ impl RandomGenerator {
         Tensorable: AsMutTensor<Element = Scalar>,
     {
-        Scalar::fill_tensor(self, UniformBoolean, output);
+        Scalar::fill_tensor(self, UniformBinary, output);
     }
 
-    /// Generates a tensor of random boolean values of a given size.
+    /// Generates a tensor of random binary values of a given size.
     ///
     /// # Example
     ///
     /// ```rust
     /// use concrete_core::math::random::RandomGenerator;
     /// use concrete_core::math::tensor::Tensor;
     /// let mut generator = RandomGenerator::new(None);
-    /// let t: Tensor<Vec<bool>> = generator.random_uniform_boolean_tensor(10);
+    /// let t: Tensor<Vec<u32>> = generator.random_uniform_binary_tensor(10);
     /// assert_eq!(t.len(), 10);
     /// ```
-    pub fn random_uniform_boolean_tensor<Scalar: RandomGenerable<UniformBoolean>>(
+    pub fn random_uniform_binary_tensor<Scalar: RandomGenerable<UniformBinary>>(
         &mut self,
         size: usize,
     ) -> Tensor<Vec<Scalar>> {
-        Scalar::generate_tensor(self, UniformBoolean, size)
+        Scalar::generate_tensor(self, UniformBinary, size)
+    }
+
+    /// Generates a random uniform ternary value.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use concrete_core::math::random::RandomGenerator;
+    /// let mut generator = RandomGenerator::new(None);
+    /// let random: u32 = generator.random_uniform_ternary();
+    /// ```
+    pub fn random_uniform_ternary<Scalar: RandomGenerable<UniformTernary>>(&mut self) -> Scalar {
+        Scalar::generate_one(self, UniformTernary)
+    }
+
+    /// Fills an `AsMutTensor` value with random ternary values.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use concrete_core::math::random::RandomGenerator;
+    /// use concrete_core::math::tensor::Tensor;
+    /// let mut generator = RandomGenerator::new(None);
+    /// let mut tensor = Tensor::allocate(1u32, 100);
+    /// generator.fill_tensor_with_random_uniform_ternary(&mut tensor);
+    /// ```
+    pub fn fill_tensor_with_random_uniform_ternary<Scalar, Tensorable>(
+        &mut self,
+        output: &mut Tensorable,
+    ) where
+        Scalar: RandomGenerable<UniformTernary>,
+        Tensorable: AsMutTensor<Element = Scalar>,
+    {
+        Scalar::fill_tensor(self, UniformTernary, output);
+    }
+
+    /// Generates a tensor of random ternary values of a given size.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use concrete_core::math::random::RandomGenerator;
+    /// use concrete_core::math::tensor::Tensor;
+    /// let mut generator = RandomGenerator::new(None);
+    /// let t: Tensor<Vec<u32>> = generator.random_uniform_ternary_tensor(10);
+    /// assert_eq!(t.len(), 10);
+    /// ```
+    pub fn random_uniform_ternary_tensor<Scalar: RandomGenerable<UniformTernary>>(
+        &mut self,
+        size: usize,
+    ) -> Tensor<Vec<Scalar>> {
+        Scalar::generate_tensor(self, UniformTernary, size)
```

Comment:
We don't need another function for Gaussian keys? It would be just `random_uniform` is that it?

Reply:
Those already exist !